ocs_ci/utility/gcp.py

```
@@ -75,6 +77,251 @@ def load_service_account_key_dict(filepath=SERVICE_ACCOUNT_KEY_FILEPATH):
+    if not match:
+        raise ValueError(f"Failed to parse WIF audience URL: {audience}")
+    params = {
+        "project_number": match.group(1),
+        "pool_id": match.group(2),
+        "provider_id": match.group(3),
+    }
+    logger.info(f"Extracted WIF params: {params}")
+    return params
+
+
+def create_noobaa_gcp_service_account(gcp_project_id, project_number, pool_id, sa_name):
```

Ruff (0.16.0)

[warning] Too many statements (54 > 50)

(PLR0915)